### ISSUE TITLE
Unable to update global tool to the latest prerelease version

### ISSUE THREAD

**Issue by danroth27** [dotnet/sdk#2551]
I tried updating to the latest prerelease version of the dotnet-httprepl tool, but there doesn't appear to be a way to do so using the update command:

```
C:\>dotnet tool update -g --add-source https://dotnet.myget.org/F/dotnet-core/api/v3/index.json dotnet-httprepl
error NU1103: Unable to find a stable package dotnet-httprepl with version
error NU1103:   - Found 9 version(s) in dotnet-core [ Nearest version: 2.2.0-preview3-35304 ]
error NU1103:   - Found 1 version(s) in nuget.org [ Nearest version: 2.2.0-preview2-35157 ]
error NU1103:   - Found 0 version(s) in Microsoft Visual Studio Offline Packages
error NU1103:   - Found 0 version(s) in C:\Program Files\dotnet\sdk\NuGetFallbackFolder
error NU1103:   - Found 0 version(s) in blazor-dev
Tool 'dotnet-httprepl' failed to update due to the following:
The tool package could not be restored.
Tool 'dotnet-httprepl' failed to install. This failure may have been caused by:

* You are attempting to install a preview release and did not use the --version option to specify the version.
* A package by this name was found, but it was not a .NET Core tool.
* The required NuGet feed cannot be accessed, perhaps because of an Internet connection problem.
* You mistyped the name of the tool.
```

Notice that the message recommends using the `--version` option, which doesn't appear to be supported with `update`:

```
C:\>dotnet tool update -g --version 2.2.0-* --add-source https://dotnet.myget.org/F/dotnet-core/api/v3/index.json dotnet-httprepl
Unrecognized command or argument '2.2.0-*'
Unrecognized command or argument 'dotnet-httprepl'
```

**Comment by danroth27** [comment]
A workaround is to uninstall and reinstall the tool.

**Comment by livarcocc** [comment]
@wli3 @peterhuene can one of you take a look?

**Comment by peterhuene** [comment]
I think `dotnet tool update` doesn't support a `--version` option currently because there's no wildcard that will get us the "latest version including prerelease" we can use for a `PackageReference`.  If we had that, I would expect a `--prerelease` option for `dotnet tool update` rather than a `--version` option, since the intent of the tool is not to install a particular version or range, but rather update to latest known (with or without considering prerelease versions).

The bug here, however, is that the help text printed is for `dotnet tool install` and not `dotnet tool update`, which supports different options obviously.  We should fix the error message because I don't see `tool update` supporting `--version`.

Until we add support for a `--prerelease` option, the workaround for the scenario would have to be to uninstall the tool and reinstall the particular desired version.

**Comment by danroth27** [comment]
> the intent of the tool is not to install a particular version or range, but rather update to latest known (with or without considering prerelease versions).

Well, I think it would be nice to have a way to update a prerelease tool without having to uninstall and install it. Please consider broadening the semantics of `update` to allow for this. My intuition was the using `--version` with `update` would work and I was surprised that it didn't. It seems like a natural extension of the `update` command. You don't always want to update to the latest version (stable or prerelease). For example, you might want to update to the latest patch of the major version you're currently using without jumping to latest due to breaking changes.

**Comment by peterhuene** [comment]
I can be convinced of having `--version` for `tool update`.  It would at least unblock this scenario and provide some consistency with `tool install`.

@wli3 what are your thoughts?

**Comment by dasMulli** [comment]
I guess the NuGet feature request needed here is https://github.com/NuGet/Home/issues/912 (needing a `*-*` version pattern for the temp project)

**Comment by danroth27** [comment]
In my case I specifically wanted to specify 2.2.0-* as the version. There are 3.0.0-* versions of the dotnet-httprepl tool, but they only work with .NET Core 3.0. I needed the latest 2.2 version.

**Comment by wli3** [comment]
fixed in https://github.com/dotnet/sdk/pull/19646
